### PR TITLE
CRIU Xtrace prepares the trace output file

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Invoke `startTraceWorkerThread()` before checking `traceMethodTable`/`triggerOnMethods`;
Added tracepoints;
Added a test.

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
jenkins test sanity xlinux jdk17

**Comment by tajila** [comment]
jenkins compile win jdk8

File: runtime/rasdump/dmpsup.c [hunk 1]
```diff
@@ -410,6 +410,7 @@ criuReloadXDumpAgents(J9JavaVM *vm, J9VMInitArgs *vmArgs)
 	IDATA result = configureDumpAgents(vm, vmArgs, FALSE);
 	unlockConfig();
 
+	Trc_trcengine_criu_criuReloadXDumpAgents_Exit(vm->mainThread, result);
```
Comment (keithc-ca) [inline runtime/rasdump/dmpsup.c:413]: This should be paired with an "Enter" tracepoint, otherwise I think it will break indentation for `-Xtrace:iprint`

Comment (tajila) [reply runtime/rasdump/dmpsup.c:413]: @JasonFengJ9 ^^

Comment (JasonFengJ9) [reply runtime/rasdump/dmpsup.c:413]: Created 
* https://github.com/eclipse-openj9/openj9/pull/17057

File: runtime/rastrace/trcengine.c [hunk 2]
```diff
@@ -1015,19 +1015,31 @@ criuRestoreInitializeTrace(J9VMThread *thr)
...
+				Trc_trcengine_criu_nomethodentries_succeed(thr);
 			}
+		} else {
+			Trc_trcengine_criu_startTraceWorkerThread_failed(thr);
 		}
+	} else {
+		Trc_trcengine_criu_traceInitializationHelper_failed(thr);
 	}
 
+	Trc_trcengine_criu_criuRestoreInitializeTrace_Exit(thr, result);
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply runtime/rastrace/trcengine.c:1042]: Added

**Pull request merged**